### ISSUE TITLE
Failure summary for "empty-table-header" is confusing for empty th

### ISSUE THREAD

**Issue by connorjclark** [dequelabs/axe-core#2956]
Product: axe-core

Expectation: 

```
<table>
  <tr>
    <th></th>
  </tr>
</table>
```

Rule should fail saying: `Element does not have text that is visible to screen readers`

Actual:

Rule is marked `incomplete` and says: `Unable to determine if element has children`

Motivation: More immediate advice.

<pre><code>
axe-core version: 4.2.1

**Comment by straker** [comment]
Figured it out. Turns out that we marked the rule to return as [Needs Review rather than failure](https://github.com/dequelabs/axe-core/blob/develop/lib/rules/empty-table-header.json#L5) when there is no text content. However, instead of using the [failure message of the check](https://github.com/dequelabs/axe-core/blob/develop/lib/checks/shared/has-visible-text.json#L8) is uses the `incomplete` message which is `Unable to determine if element has children`.

It's important to note that we do want the rule to be marked as incomplete and not fail as there are legit reasons a table header could be empty. See https://github.com/dequelabs/axe-core/issues/2604

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base,
Earlier it was displaying as  `Unable to determine if element has children`
Now it is appearing as `Element does not have text that is visible to screen readers`
Before changes:
![image](https://user-images.githubusercontent.com/47409579/124442956-e022a100-dd9a-11eb-953e-22ca82ea9e9b.png)

after changes:
![image](https://user-images.githubusercontent.com/47409579/124443247-2546d300-dd9b-11eb-8f7a-183b525f97b0.png)
